### PR TITLE
HPCC-22415 Remove redundant security checking from dali

### PR DESCRIPTION
**Pull request by ghalliday**
Signed-off-by: Gavin Halliday <gavin.halliday@lexisnexis.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [x] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

**Comment by ghalliday** [comment]
@RussWhitehead and @jakesmith please review.

File: dali/base/dasess.cpp [hunk 1]
```diff
@@ -986,21 +986,10 @@ class CClientSessionManager: public CSessionManagerBase, implements IConnectionM
...
-                }
-            }
-            else
-            {
-                mb.append(reqSignature);
-                reqUTCTimestamp.serialize(mb);
-            }
+            //Backwards compatibility in case another parameter is added later, otherwise could be removed
+            CDateTime reqUTCTimestamp;
+            mb.append("");
```
Comment (jakesmith) [inline dali/base/dasess.cpp]: trivial: worth a comment saying what legacy serialized field this was // base64 signature.

**Comment by jakesmith** [comment]
@ghalliday - looks ok afaics. I think it would be worth adding some notes to the JIRA descriptor and PR commit message to elaborate on why removing, e.g. whitelist..

@RussWhitehead - tagging you for review.

**Comment by RussWhitehead** [comment]
@ghalliday The changes look reasonable, but I am not clear why this code is no longer necessary. Can you update the Jira or PR?  Also @afishbeck should be aware of this change

**Comment by ghalliday** [comment]
@RussWhitehead the question is what do we gain from this code being present?  I'll update the jira, but if you can think of a good reason for keeping it then it should stay.

**Comment by ghalliday** [comment]
@jakesmith I had some local changes I had forgot to delete - included in the second commit.

**Comment by jakesmith** [comment]
@ghalliday - createDaliSignature not called now, so looks fine.

Tagging @RussWhitehead - for Gavin's question above.

File: dali/server/daldap.cpp [hunk 2]
```diff
@@ -147,80 +147,10 @@ class CDaliLdapConnection: implements IDaliLdapConnection, public CInterface
...
-                pDSM = queryDigitalSignatureManagerInstanceFromEnv();
-            if (pDSM && pDSM->isDigiVerifierConfigured())
-            {
-                StringBuffer requestTimestamp;
-                reqUTCTimestamp.getString(requestTimestamp, false);//extract timestamp string from Dali request
-
-                CDateTime now;
-                now.setNow();
-                CDateTime daliTime(now);
-                if (requestSignatureAllowedClockVarianceSeconds)//allow for clock variance between machines
```
Comment (RussWhitehead) [inline dali/server/daldap.cpp]: Need to remove this feature from the configuration

**Comment by ghalliday** [comment]
@RussWhitehead thanks for the reminder.  Please check the latest commit.

**Review by RussWhitehead — COMMENTED** [review]
@ghalliday Looks good, but what happens when a client engine with your changes calls a Dali running an older 7.x, which expects a signature in the request?

**Comment by ghalliday** [comment]
@RussWhitehead 
Good question.  I think if the signature and the password are blank then the old code would treat the user as authenticated.  (Please can you double check my logic.) 

There might be a problem if a non-workunit token password was passed to an old dali.   That shouldn't ever happen when connecting from a 7.x system since the password isn't stored.
I will double check though - please can you also check the logic.

**Comment by RussWhitehead** [comment]
@ghalliday It looks like you are correct, a missing signed token will be ignored by 7.2.x.   I do recommend testing it since a failure will cause production failures

**Comment by ghalliday** [comment]
@jakesmith do you have a simple way of testing this?

**Comment by jakesmith** [comment]
@ghalliday - doing : 
```
daliadmin <dali-ip> dfsverify <logicalfilename>
```

with theses changes to a 7.2.x dali should test it.

**Comment by ghalliday** [comment]
@richardkchapman I have checked the command that Jake suggested worked.  Please review carefully.

**Comment by richardkchapman** [comment]
@afishbeck Are you happy with this?

**Comment by afishbeck** [comment]
@richardkchapman yes, Russ and I had discussed this offline.  We concluded the premise of this JIRA was correct.  This checking wasn't providing benefit.  Whitelists and in the future PKI based component trust will provide better intra-component security.  End user authentication and permissions will happen at ESP.

**Pull request merged**